refactor(cms): persist essay answer likes count

nickhsine opened the pull request:
## Summary
Persist likesCount as an integer field and sync counts through PostEssayAnswerLike hooks.

## Changes
- Replace virtual likesCount with an integer column and hide it from mutations
- Add afterOperation hook on PostEssayAnswerLike to atomically inc/dec counts via raw SQL with GraphQLError handling
- Add notes around atomic updates and transaction scope for to-one disconnect handling
- Prisma migration to add the likesCount column.

## Additional Notes
There is a TODO note in this PR. If the afterOperation hook fails to adjust counts, enqueue a PubSub message to recompute the latest likesCount so data stays in sync.

nickhsine committed:
refactor(cms): persist essay answer likes and sync via like hooks

- replace PostEssayAnswer.likesCount virtual with integer field for ordering/filtering and hide it from mutations
- add afterOperation on PostEssayAnswerLike to atomically inc/dec likes via raw SQL with GraphQLError detection
- update Prisma/GraphQL schema to include the new likesCount column and order/filter inputs
- add migrations/20251201153330_add_likes_count_in_post_essay_answer/migration.sql

nickhsine commented on packages/cms/lists/post-essay-answer-like.ts:
```diff
@@ -130,5 +131,71 @@ export default list<ListType<'PostEssayAnswerLike'>>({
+        await Promise.all([
+          incrementLikesCount(answerConnectId),
+          decrementLikesCount(answerDisconnectId),
+        ])
```
fixed.

nickhsine commented on packages/cms/lists/post-essay-answer-like.ts:
```diff
@@ -130,5 +131,71 @@ export default list<ListType<'PostEssayAnswerLike'>>({
+          const debugInfo = result.extensions?.debug as
+            | { message?: string }
+            | undefined
+          const errorMsg = debugInfo?.message?.trim() ?? ''
```
fixed.

nickhsine commented on packages/cms/lists/post-essay-answer-like.ts:
```diff
@@ -130,5 +131,71 @@ export default list<ListType<'PostEssayAnswerLike'>>({
+        if (!answerId) return
+        // Atomic row update in Postgres to avoid race conditions on concurrent unlikes
+        const result = await context.prisma
+          .$executeRaw`UPDATE "PostEssayAnswer" SET "likesCount" = "likesCount" - 1 WHERE "id" = ${Number(answerId)}`
```
fixed.

nickhsine commented on packages/cms/schema.prisma:
```diff
@@ -421,6 +421,7 @@ model PostEssayAnswer {
   member                          Member?               @relation("PostEssayAnswer_member", fields: [memberId], references: [id])
   memberId                        String?               @map("member")
   content                         String                @default("")
+  likesCount                      Int?                  @default(0)
```
fixed

ericsen-tsai commented on packages/cms/lists/post-essay-answer-like.ts:
```diff
@@ -130,5 +131,75 @@ export default list<ListType<'PostEssayAnswerLike'>>({
+        await Promise.all([
+          incrementLikesCount(answerConnectId),
+          decrementLikesCount(answerDisconnectId),
+        ])
```
question:
Do we have to handle this situation (user can only like and unlike, that is, create and delete)?
So maybe we can just omit the update mutation for this list?

nickhsine replied:
fixed in [feat(cms): remove update handling for answer likes](https://github.com/kids-reporter/kids-reporter-monorepo/pull/851/commits/b3295d179936a9c4dee92f6af7b6b544fc23f67e)

ericsen-tsai reviewed (approved):
LGTM

nickhsine committed:
feat(cms): remove update handling for answer likes

- Omit the PostEssayAnswerLike update mutation and deny update operations at the access layer.
- Drop update-specific resolveInput checks and afterOperation increment/decrement logic since likes are only created or deleted.
- Delete updatePostEssayAnswerLike and updatePostEssayAnswerLikes mutation in schema.graphql
